src/features/user/MFV2/ForestProgress/microComponents/ProgressData.tsx

```diff
@@ -0,0 +1,135 @@
+import {
+  TreesPlantedIcon,
+  AreaRestoredIcon,
+  ConservedAreaIcon,
+} from '../../../../../../public/assets/images/icons/ProgressBarIcons';
+import { DataType } from '../ForestProgressItem';
+import { useTranslations } from 'next-intl';
+import { useMemo } from 'react';
+import styles from '../ForestProgress.module.scss';
+import StackedBarGraph from './StackedBarGraph';
+
+export interface ProgressDataProps {
+  giftPercentage: number;
+  personalPercentage: number;
+  gift: number;
+  personal: number;
+  dataType: DataType;
+  target: number;
+}
+
+const ForestProgressIcon = ({ dataType }: { dataType: DataType }) => {
+  switch (dataType) {
+    case 'treesPlanted':
+      return <TreesPlantedIcon width={19} />;
+    case 'areaRestored':
+      return <AreaRestoredIcon width={17} />;
+    case 'areaConserved':
+      return <ConservedAreaIcon width={13} />;
+  }
+};
+
+const GiftReceivedFromCommunity = ({
+  gift,
+  label,
+}: {
+  gift: number;
+  label: string;
+}) => {
+  return (
+    <>
+      {' '}
+      {gift !== undefined && gift > 0 && (
+        <div className={styles.communityReceived}>{label}</div>
+      )}
+    </>
+  );
+};
```

Comment:
I feel like this abstraction was unnecessary. You could simply include this logic within `ProgressData` itself.